### PR TITLE
[phi::DenseTensor] Replace Tensor with phi::DenseTensor

### PR DESCRIPTION
**Pull request by Liyulingyue**
### PR types

Others 
### PR changes

Others 
### Describe

Replace Tensor with phi::DenseTensor.
According to https://github.com/PaddlePaddle/Paddle/pull/48525 , the action will not happen in any file with 'Opmaker'.

### CODE REVIEW

File: paddle/fluid/imperative/gradient_accumulator.cc [hunk 1]
```diff
@@ -155,7 +155,7 @@ TType* GetEmptyInnerTensor(paddle::experimental::Tensor* dst) {
       dst->defined(),
       false,
       platform::errors::Fatal(
-          "The underlying Tensor implementation should be nullptr"));
```
Comment (zyfncg) [inline paddle/fluid/imperative/gradient_accumulator.cc]: imperative目录中已经开始使用上层Tensor的概念了，此文件中的这些注释内容目前不用修改

Comment (Liyulingyue) [reply paddle/fluid/imperative/gradient_accumulator.cc]: > imperative目录中已经开始使用上层Tensor的概念了，此文件中的这些注释内容目前不用修改

done. everything resumed

**Review by YuanRisheng — APPROVED** [review]

**Review by Ligoml — APPROVED** [review]
LGTM

**Review by XiaoguangHu01 — APPROVED** [review]
LGTM

**Pull request merged**